crates/compiler/gen_llvm/src/llvm/lowlevel.rs

```diff
@@ -1172,9 +1172,69 @@ pub(crate) fn run_low_level<'a, 'ctx>(
             // which could be useful to look at when implementing this.
             todo!("implement checked float conversion");
         }
-        I128OfDec => {
-            arguments!(dec);
-            dec_unary_op(env, bitcode::DEC_TO_I128, dec)
+        NumWithoutDecimalPoint | NumWithDecimalPoint => {
+            // Dec uses an I128 under the hood, so no conversion is needed.
+            arguments!(arg);
+            arg
+        }
+        NumF32ToParts => {
+            arguments!(arg);
+            let fn_name = bitcode::NUM_F32_TO_PARTS;
+            call_bitcode_fn_and_cast_result(env, &[arg], fn_name, layout_interner, layout)
+        }
+        NumF64ToParts => {
+            arguments!(arg);
+            let fn_name = bitcode::NUM_F64_TO_PARTS;
+            call_bitcode_fn_and_cast_result(env, &[arg], fn_name, layout_interner, layout)
+        }
+        NumF32FromParts => {
+            arguments!(arg);
+            let fn_name = bitcode::NUM_F32_FROM_PARTS;
+
+            let roc_call_alloca = env
+                .builder
+                .new_build_alloca(arg.get_type(), "roc_call_alloca");
+            env.builder.new_build_store(roc_call_alloca, arg);
+
+            let fn_val = env.module.get_function(fn_name).unwrap();
+            let zig_param_type = fn_val.get_params()[0].get_type();
+            let zig_value =
+                env.builder
+                    .new_build_load(zig_param_type, roc_call_alloca, "zig_value");
+
+            call_bitcode_fn(env, &[zig_value], fn_name)
+        }
+        NumF64FromParts => {
+            arguments!(arg);
+            let fn_name = bitcode::NUM_F64_FROM_PARTS;
+
+            let roc_call_alloca = env
+                .builder
+                .new_build_alloca(arg.get_type(), "roc_call_alloca");
+            env.builder.new_build_store(roc_call_alloca, arg);
+
+            let fn_val = env.module.get_function(fn_name).unwrap();
+
+            let zig_params_types: Vec<_> =
+                fn_val.get_params().iter().map(|p| p.get_type()).collect();
+
+            let zig_record_type = env.context.struct_type(&zig_params_types, false);
+            let zig_recode_value = env
+                .builder
+                .new_build_load(zig_record_type, roc_call_alloca, "zig_value")
+                .into_struct_value();
+
+            let zig_value_0 = env
+                .builder
+                .build_extract_value(zig_recode_value, 0, "zig_value_0")
+                .unwrap();
+
+            let zig_value_1 = env
+                .builder
+                .build_extract_value(zig_recode_value, 1, "zig_value_1")
+                .unwrap();
```

Comment:
Why is this split in two parts?

Reply:
Only way I could make it work. But this seems to fail for macos M1.

Reply:
Sorry that I have been so slow on reviewing and helping here. Just haven't have time for development lately. Anyway, this looks to be hitting some abi fun.
On x86_64:
```
double @roc_builtins.num.f64_from_parts(i64 %0, i24 %1)
```
On Aarch64:
```
double @roc_builtins.num.f64_from_parts([2 x i64] %0)
```
On x86_32:
```
double @roc_builtins.num.f64_from_parts(%num.F64Parts %0)
```
On x86_64 windows:
```
double @roc_builtins.num.f64_from_parts(ptr nocapture noundef readonly %0)
```
On wasm32:
```
double @roc_builtins.num.f64_from_parts(ptr nocapture nonnull readonly align 8 %0)
```

So these fall into three categories:

For x86_64: it needs to be passed as you have specified

For Aarch64, x86_32: it needs to be passed the same way as `NumF32FromParts` currently is.

For windows and wasm32: it needs to be passed by pointer. So like `NumF32FromParts`, but instead of loading the value, pass in a pointer to the alloca.

Reply:
Good test for llvm wasm and the pointer handling being correct: `cargo test-gen-llvm-wasm -- gen_num::f64`

Reply:
oh, also, all of these correct function signatures came from looking at the `.ll` files that get generated in `crates/compiler/builtins/bitcode/zig-out/builtins-*.ll`

Reply:
Oh, one last note, you may want wait for https://github.com/roc-lang/roc/pull/6610 to merge and rebase on top of it. Should be in very shortly. It changes how we check targets for making decisions like I mentioned are needed above. Should be in hopefully in an hour or so 🤞